server.go

```diff
@@ -110,6 +110,15 @@ func (srv *Server) Apply(req *Request) (*Responder, error) {
 	return res, err
 }
 
+func (srv *Server) Close() {
+	srv.mutex.Lock()
```

Comment:
actually, The only way to close server is using Close() method in another thread.